pkg/flow/internal/dag/dag.go

```diff
@@ -77,6 +80,18 @@ func (g *Graph) Add(n Node) {
 // the graph.
 func (g *Graph) GetByID(id string) Node { return g.nodeByID[id] }
 
+// GetByIDPrefix returns a slice of nodes with an ID Prefix.
+func (g *Graph) GetByIDPrefix(idPrefix string) []Node {
```

Comment:
This is very inefficient and not how I think we should do this but I made a go at it using just the graph for prototyping purposes. Let me know what you think @rfratto now that the rest of the code has moved in the direction you are looking for.

1 alternative is to track the export/argument ids as they are added then pass those in to the config.go funcs instead.

Another alternative is to implement a tree that is optimized for prefix searching, ie. https://github.com/tchap/go-patricia/